### ISSUE TITLE
Bot Consistently Crashing "Failed To Get Meeting Members"

### ISSUE THREAD

**Issue by dborowicz16** [TobiasPankner/Teams-Auto-Joiner#82]
Hello! I love this bot you have created. I am trying to repurpose this bot to function to be "always running" to act as a meeting room for people to chat. The bot joins meetings flawlessly, however when the bot is inside of the meeting two things generally happen. The first is that the bot gets stuck on an endless loop of searching for members in the meeting but failing to do so. When it fails to find the meeting members it prints "Failed to get meeting members" but the bot does not crash. It just seems to be stuck in the meeting forever. The other thing that will happen is that it does find how many member's are in the meeting. When it does, the bot will crash immediately.

Here is the output provided by the terminal when the bot crashes:

```
[WDM] - Current google-chrome version is 88.0.4324
[WDM] - Get LATEST driver version for 88.0.4324
[WDM] - Driver [C:\Users\TeamsBot\.wdm\drivers\chromedriver\win32\88.0.4324.96\chromedriver.exe] found in cache

DevTools listening on ws://127.0.0.1:51225/devtools/browser/06ce2637-d224-42c6-a52b-ed169352cb01
[4972:12424:0311/095929.001:ERROR:device_event_log_impl.cc(211)] [09:59:28.999] USB: usb_device_handle_win.cc:1049 Failed to read descriptor from node connection: A device attached to the system is not functioning. (0x1F)
[4972:12424:0311/095929.013:ERROR:device_event_log_impl.cc(211)] [09:59:29.010] USB: usb_device_handle_win.cc:1049 Failed to read descriptor from node connection: A device attached to the system is not functioning. (0x1F)
Resized window width
Resized window height
Found page, do not click anything on the webpage from now on.

Testing The Water Cooler
        General

[09:59:56] Looking for new meetings

[10:00:08] Looking for new meetings

[10:00:19] Looking for new meetings

[10:00:31] Looking for new meetings

[10:00:42] Looking for new meetings

[10:00:54] Looking for new meetings
Found meetings:
        Testing The Water Cooler -> General 1615473512 [Channel]
[1516:2036:0311/100059.001:ERROR:audio_rtp_receiver.cc(90)] AudioRtpReceiver::OnSetVolume: No audio channel exists.
Joined meeting: Testing The Water Cooler -> General
Failed to get meeting members
[1516:2036:0311/100102.662:ERROR:internal_decoder_factory.cc(59)] Trying to create decoder for unsupported format
[1516:2036:0311/100102.667:ERROR:internal_decoder_factory.cc(59)] Trying to create decoder for unsupported format
[1516:2036:0311/100102.691:ERROR:internal_decoder_factory.cc(59)] Trying to create decoder for unsupported format
[1516:2036:0311/100102.696:ERROR:internal_decoder_factory.cc(59)] Trying to create decoder for unsupported format
[1516:2036:0311/100102.719:ERROR:internal_decoder_factory.cc(59)] Trying to create decoder for unsupported format
[1516:2036:0311/100102.724:ERROR:internal_decoder_factory.cc(59)] Trying to create decoder for unsupported format
[1516:2036:0311/100102.748:ERROR:internal_decoder_factory.cc(59)] Trying to create decoder for unsupported format
[1516:2036:0311/100102.752:ERROR:internal_decoder_factory.cc(59)] Trying to create decoder for unsupported format
[1516:2036:0311/100102.775:ERROR:internal_decoder_factory.cc(59)] Trying to create decoder for unsupported format
[1516:2036:0311/100102.780:ERROR:internal_decoder_factory.cc(59)] Trying to create decoder for unsupported format
[1516:2036:0311/100102.801:ERROR:internal_decoder_factory.cc(59)] Trying to create decoder for unsupported format
[1516:2036:0311/100102.806:ERROR:internal_decoder_factory.cc(59)] Trying to create decoder for unsupported format
[1516:2036:0311/100102.829:ERROR:internal_decoder_factory.cc(59)] Trying to create decoder for unsupported format
[1516:2036:0311/100102.834:ERROR:internal_decoder_factory.cc(59)] Trying to create decoder for unsupported format
[1516:2036:0311/100102.839:ERROR:webrtc_video_engine.cc(3335)] Absent receive stream; ignoring clearing encoded frame sink for ssrc 0
[1516:2036:0311/100102.839:ERROR:webrtc_video_engine.cc(3335)] Absent receive stream; ignoring clearing encoded frame sink for ssrc 0
[1516:2036:0311/100102.840:ERROR:webrtc_video_engine.cc(3335)] Absent receive stream; ignoring clearing encoded frame sink for ssrc 0
[1516:2036:0311/100102.843:ERROR:webrtc_video_engine.cc(3335)] Absent receive stream; ignoring clearing encoded frame sink for ssrc 0
[1516:2036:0311/100102.844:ERROR:webrtc_video_engine.cc(3335)] Absent receive stream; ignoring clearing encoded frame sink for ssrc 0
[1516:2036:0311/100102.845:ERROR:webrtc_video_engine.cc(3335)] Absent receive stream; ignoring clearing encoded frame sink for ssrc 0
[1516:2036:0311/100102.845:ERROR:webrtc_video_engine.cc(3335)] Absent receive stream; ignoring clearing encoded frame sink for ssrc 0

[10:01:11] Meeting search is paused because you are still in a meeting

[10:01:25] Meeting search is paused because you are still in a meeting

[10:01:39] Meeting search is paused because you are still in a meeting
[OpenH264] this = 0x000002B10601E850, Warning:[Rc] iDid = 0,iContinualSkipFrames(3) is large
[OpenH264] this = 0x000002B10601E850, Warning:[Rc] iDid = 0,iContinualSkipFrames(6) is large
[OpenH264] this = 0x000002B10601E850, Warning:[Rc] iDid = 0,iContinualSkipFrames(9) is large
[OpenH264] this = 0x000002B10601E850, Warning:[Rc] iDid = 0,iContinualSkipFrames(12) is large
[OpenH264] this = 0x000002B10601E850, Warning:[Rc] iDid = 0,iContinualSkipFrames(15) is large
[OpenH264] this = 0x000002B10601E850, Warning:[Rc] iDid = 0,iContinualSkipFrames(18) is large
[OpenH264] this = 0x000002B10601E850, Warning:[Rc] iDid = 0,iContinualSkipFrames(21) is large
[OpenH264] this = 0x000002B103BB7570, Warning:Actual input framerate 0.000000 is different from framerate in setting 29.000000, suggest to use other rate control modes
[OpenH264] this = 0x000002B1056F98B0, Warning:Actual input framerate 0.000000 is different from framerate in setting 30.000000, suggest to use other rate control modes
[1516:14180:0311/100139.889:ERROR:stun_port.cc(96)] Binding request timed out from 192.168.164.x:65072 ({70B073FE-05B3-45B2-A154-C8EBF8EE5A67})

[10:01:53] Meeting search is paused because you are still in a meeting

[10:02:08] Meeting search is paused because you are still in a meeting
Current: 2
Total: 2
[OpenH264] this = 0x000002B108C7FA70, Warning:Actual input framerate 0.000000 is different from framerate in setting 30.000000, suggest to use other rate control modes
[OpenH264] this = 0x000002B108C80550, Warning:Actual input framerate 0.000000 is different from framerate in setting 29.000000, suggest to use other rate control modes
[1516:14180:0311/100212.383:ERROR:stun_port.cc(308)] Port[8b45050:audio:1:0:local:Net[{88661D32-D9E2-472C-BD47-52456D928ADF}:10.1.100.x/24:Ethernet:id=2]]: UDP send of 39 bytes failed with error 10053
[1516:14180:0311/100212.445:ERROR:stun_port.cc(308)] Port[8b45050:audio:1:0:local:Net[{88661D32-D9E2-472C-BD47-52456D928ADF}:10.1.100.x/24:Ethernet:id=2]]: UDP send of 98 bytes failed with error 10053
Traceback (most recent call last):
  File "C:\Users\TeamsBot\Downloads\Teams-Auto-Joiner-master\Teams-Auto-Joiner-master\auto_joiner.py", line 689, in <module>
    main()
  File "C:\Users\TeamsBot\Downloads\Teams-Auto-Joiner-master\Teams-Auto-Joiner-master\auto_joiner.py", line 664, in main
    if handle_leave_threshold(members_count, total_members):
  File "C:\Users\TeamsBot\Downloads\Teams-Auto-Joiner-master\Teams-Auto-Joiner-master\auto_joiner.py", line 533, in handle_leave_threshold
    if 0 < float(leave_percentage) <= 150:
ValueError: could not convert string to float: ''
```

Additionally, here is my config file:

```
{
  "email": "",
  "password": "",

  "run_at_time": "",

  "meeting_mode": 1,
  "organisation_num" : 1,
  "random_delay": false,
  "check_interval": 10,

  "join_message": "",

  "auto_leave_after_min": -1,
  "leave_if_last": true,
  "leave_threshold_number": "",
  "leave_threshold_percentage": "",
  "pause_search": true,

  "headless" : false,
  "mute_audio": false,
  "chrome_type": "google-chrome",

  "blacklist": [
    {
      "team_name": "Example Team 1",
      "channel_names": [
        "First Channel to blacklist in this team",
        "Second Channel to blacklist"
      ]
    },
     {
      "team_name": "Blacklist whole team",
      "channel_names": [
      ]
    }
  ],
  "blacklist_meeting_re": ""
}
```

And finally the only change I've made to the auto_joiner.py would be commenting out the block of code that mutes audio/video so that the video/audio would be constantly on. The change I have made to this file is shown here:

```
# turn camera off
    #video_btn = browser.find_element_by_css_selector("toggle-button[data-tid='toggle-video']>div>button")
    #video_is_on = video_btn.get_attribute("aria-pressed")
    #if video_is_on == "true":
    #    video_btn.click()
    #    print("Video off")

    # turn mic off
    #audio_btn = browser.find_element_by_css_selector("toggle-button[data-tid='toggle-mute']>div>button")
    #audio_is_on = audio_btn.get_attribute("aria-pressed")
    #if audio_is_on == "true":
    #    audio_btn.click()
    #    print("Audio off")
```

If you could help me out with this crashing error and get the bot to run consistently to suit my needs I would greatly appreciate it!

**Comment by Leoyut** [comment]
Same problem brother

**Comment by Wojtulasek** [comment]
I changed the 525 line and it didn't work. Can you help?

**Comment by Leoyut** [comment]
Im also facing the same issue,after updating the code ........

**Comment by TobiasPankner** [comment]
Please provide the console log and make sure you are on the newest version.

**Comment by DishantX** [comment]
> Please provide the console log and make sure you are on the newest version.

newest version of what?

**Comment by TobiasPankner** [comment]
Newest version of this program. (Teams-Auto-Joiner)

**Comment by DishantX** [comment]
Similar Issue after updating the tools

Crash Logs:
```

C:\Users\dcmx\Videos\Captures\Teams-Auto-Joiner-master>python auto_joiner.py

[WDM] - Current google-chrome version is 91.0.4472
[WDM] - Get LATEST driver version for 91.0.4472

DevTools listening on ws://127.0.0.1:65259/devtools/browser/0870234b-bc6a-4afa-958e-5cf11028348d
[3660:9052:0617/082451.445:ERROR:device_event_log_impl.cc(214)] [08:24:51.445] USB: usb_device_handle_win.cc:1058 Failed to read descriptor from node connection: A device attached to the system is not functioning. (0x1F)
[3660:9052:0617/082451.452:ERROR:device_event_log_impl.cc(214)] [08:24:51.445] USB: usb_service_win.cc:352 Could not get child device's service name: Element not found. (0x490)
[3660:9052:0617/082451.454:ERROR:device_event_log_impl.cc(214)] [08:24:51.455] USB: usb_device_handle_win.cc:1058 Failed to read descriptor from node connection: A device attached to the system is not functioning. (0x1F)
Resized window width
Resized window height
Found page, do not click anything on the webpage from now on.

12A SCIENCE(NEW 21-22)
        CHEMISTRY [MEETING]

[08:25:54] Looking for new meetings
Found meetings:
        12A SCIENCE(NEW 21-22) -> CHEMISTRY 1623687519 [Channel]
        Class 12A Chemistry from 7:00 AM to 9:00 AM 1623893400 [Calendar]

Video off
Audio off
[8708:632:0617/082606.641:ERROR:audio_rtp_receiver.cc(95)] AudioRtpReceiver::OnSetVolume: No audio channel exists.
Joined meeting: Class 12A Chemistry from 7:00 AM to 7:40 AM
[8708:632:0617/082608.311:ERROR:webrtc_video_engine.cc(3415)] Absent receive stream; ignoring clearing encoded frame sink for ssrc 0
[8708:632:0617/082608.312:ERROR:webrtc_video_engine.cc(3415)] Absent receive stream; ignoring clearing encoded frame sink for ssrc 0
[8708:632:0617/082608.314:ERROR:webrtc_video_engine.cc(3415)] Absent receive stream; ignoring clearing encoded frame sink for ssrc 0

[08:26:22] Meeting search is paused because you are still in a meeting
Failed to get meeting members

[08:26:34] Meeting search is paused because you are still in a meeting
Failed to get meeting members

[08:26:46] Meeting search is paused because you are still in a meeting
Failed to get meeting members
[3064:4432:0617/082651.077:ERROR:gpu_init.cc(440)] Passthrough is not supported, GL is disabled

Traceback (most recent call last):
  File "C:\Users\dcmx\Videos\Captures\Teams-Auto-Joiner-master\auto_joiner.py", line 749, in <module>
    main()
  File "C:\Users\dcmx\Videos\Captures\Teams-Auto-Joiner-master\auto_joiner.py", line 718, in main
    members_count = get_meeting_members()
  File "C:\Users\dcmx\Videos\Captures\Teams-Auto-Joiner-master\auto_joiner.py", line 490, in get_meeting_members
    meeting_elems = browser.find_elements_by_css_selector(".one-call")
  File "C:\Users\dcmx\AppData\Local\Programs\Python\Python39\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 614, in find_elements_by_css_selector
    return self.find_elements(by=By.CSS_SELECTOR, value=css_selector)
  File "C:\Users\dcmx\AppData\Local\Programs\Python\Python39\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 1005, in find_elements
    return self.execute(Command.FIND_ELEMENTS, {
  File "C:\Users\dcmx\AppData\Local\Programs\Python\Python39\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "C:\Users\dcmx\AppData\Local\Programs\Python\Python39\lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: chrome not reachable
  (Session info: chrome=91.0.4472.106)
```
Config File: 

{
  "email": "",
  "password": "",

  "run_at_time": "",

  "meeting_mode": 1,
  "organisation_num" : -1,
  "random_delay": false,
  "check_interval": 10,

  "join_message": "",

  "auto_leave_after_min": -1,
  "leave_if_last": true,
  "leave_threshold_number": "7",
  "leave_threshold_percentage": "",
  "pause_search": true,

  "headless" : false,
  "mute_audio": true,
  "chrome_type": "google-chrome",

  "blacklist": [
    {
      "team_name": "Example Team 1",
      "channel_names": [
        "First Channel to blacklist in this team",
        "Second Channel to blacklist"
      ]
    },
     {
      "team_name": "Blacklist whole team",
      "channel_names": [
      ]
    }
  ],
  "blacklist_meeting_re": "",

  "discord_webhook_url" : ""
}

**Comment by DishantX** [comment]
Similar Issue after updating the tools

Crash Logs:
```

C:\Users\dcmx\Videos\Captures\Teams-Auto-Joiner-master>python auto_joiner.py

[WDM] - Current google-chrome version is 91.0.4472
[WDM] - Get LATEST driver version for 91.0.4472

DevTools listening on ws://127.0.0.1:65259/devtools/browser/0870234b-bc6a-4afa-958e-5cf11028348d
[3660:9052:0617/082451.445:ERROR:device_event_log_impl.cc(214)] [08:24:51.445] USB: usb_device_handle_win.cc:1058 Failed to read descriptor from node connection: A device attached to the system is not functioning. (0x1F)
[3660:9052:0617/082451.452:ERROR:device_event_log_impl.cc(214)] [08:24:51.445] USB: usb_service_win.cc:352 Could not get child device's service name: Element not found. (0x490)
[3660:9052:0617/082451.454:ERROR:device_event_log_impl.cc(214)] [08:24:51.455] USB: usb_device_handle_win.cc:1058 Failed to read descriptor from node connection: A device attached to the system is not functioning. (0x1F)
Resized window width
Resized window height
Found page, do not click anything on the webpage from now on.

12A SCIENCE(NEW 21-22)
        CHEMISTRY [MEETING]

[08:25:54] Looking for new meetings
Found meetings:
        12A SCIENCE(NEW 21-22) -> CHEMISTRY 1623687519 [Channel]
        Class 12A Chemistry from 7:00 AM to 9:00 AM 1623893400 [Calendar]

Video off
Audio off
[8708:632:0617/082606.641:ERROR:audio_rtp_receiver.cc(95)] AudioRtpReceiver::OnSetVolume: No audio channel exists.
Joined meeting: Class 12A Chemistry from 7:00 AM to 7:40 AM
[8708:632:0617/082608.311:ERROR:webrtc_video_engine.cc(3415)] Absent receive stream; ignoring clearing encoded frame sink for ssrc 0
[8708:632:0617/082608.312:ERROR:webrtc_video_engine.cc(3415)] Absent receive stream; ignoring clearing encoded frame sink for ssrc 0
[8708:632:0617/082608.314:ERROR:webrtc_video_engine.cc(3415)] Absent receive stream; ignoring clearing encoded frame sink for ssrc 0

[08:26:22] Meeting search is paused because you are still in a meeting
Failed to get meeting members

[08:26:34] Meeting search is paused because you are still in a meeting
Failed to get meeting members

[08:26:46] Meeting search is paused because you are still in a meeting
Failed to get meeting members
[3064:4432:0617/082651.077:ERROR:gpu_init.cc(440)] Passthrough is not supported, GL is disabled

Traceback (most recent call last):
  File "C:\Users\dcmx\Videos\Captures\Teams-Auto-Joiner-master\auto_joiner.py", line 749, in <module>
    main()
  File "C:\Users\dcmx\Videos\Captures\Teams-Auto-Joiner-master\auto_joiner.py", line 718, in main
    members_count = get_meeting_members()
  File "C:\Users\dcmx\Videos\Captures\Teams-Auto-Joiner-master\auto_joiner.py", line 490, in get_meeting_members
    meeting_elems = browser.find_elements_by_css_selector(".one-call")
  File "C:\Users\dcmx\AppData\Local\Programs\Python\Python39\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 614, in find_elements_by_css_selector
    return self.find_elements(by=By.CSS_SELECTOR, value=css_selector)
  File "C:\Users\dcmx\AppData\Local\Programs\Python\Python39\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 1005, in find_elements
    return self.execute(Command.FIND_ELEMENTS, {
  File "C:\Users\dcmx\AppData\Local\Programs\Python\Python39\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 321, in execute
    self.error_handler.check_response(response)
  File "C:\Users\dcmx\AppData\Local\Programs\Python\Python39\lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: chrome not reachable
  (Session info: chrome=91.0.4472.106)
```
Config File: 

```
{
  "email": "",
  "password": "",

  "run_at_time": "",

  "meeting_mode": 1,
  "organisation_num" : -1,
  "random_delay": false,
  "check_interval": 10,

  "join_message": "",

  "auto_leave_after_min": -1,
  "leave_if_last": true,
  "leave_threshold_number": "7",
  "leave_threshold_percentage": "",
  "pause_search": true,

  "headless" : false,
  "mute_audio": true,
  "chrome_type": "google-chrome",

  "blacklist": [
    {
      "team_name": "Example Team 1",
      "channel_names": [
        "First Channel to blacklist in this team",
        "Second Channel to blacklist"
      ]
    },
     {
      "team_name": "Blacklist whole team",
      "channel_names": [
      ]
    }
  ],
  "blacklist_meeting_re": "",

  "discord_webhook_url" : ""
}
```

**Comment by TobiasPankner** [comment]
Please try the newest version, should be fixed with commit 9e94110
